src/structures/idt.rs

```diff
@@ -593,17 +593,51 @@ impl<T> PartialEq for Entry<T> {
 }
 
 /// A handler function for an interrupt or an exception without error code.
+#[cfg(feature = "abi_x86_interrupt")]
 pub type HandlerFunc = extern "x86-interrupt" fn(InterruptStackFrame);
+/// This type is a dummy.
+///
+/// This type alias is not used for builds which do not use the `abi_x86_interrupt` feature.
```

Comment:
Here (and for all the other `Func` types), we shouldn't have different doc comments depending on which features are enabled. These comments won't be seen on [docs.rs](https://docs.rs/). Instead, we should have a comment in the `x86_64::structures::idt` module level documentation, explaining the situation.